runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -5253,15 +5258,31 @@ class INTERPRETER_CLASS
 		_currentThread->callOutCount += 1;
 #endif /* JAVA_SPEC_VERSION >= 19 */
 		updateVMStruct(REGISTER_ARGS);
+#if JAVA_SPEC_VERSION >= 21
+		/* Remove VMaccess in the trivial downcall to benefit the JIT for better performance. */
+		if (!_currentThread->isInTrivialDownCall) {
+			VM_VMAccess::inlineExitVMToJNI(_currentThread);
+		}
+#else /* JAVA_SPEC_VERSION >= 21 */
 		VM_VMAccess::inlineExitVMToJNI(_currentThread);
+#endif /* JAVA_SPEC_VERSION >= 21 */
```

Comment:
Rather than  duplicate code, the call can just be conditional. I'm not sure the comment explains the intended behavior very well. Perhaps this?
```c
#if JAVA_SPEC_VERSION >= 21
		/* Only exit to JNI for non-trivial downcalls for better JIT performance. */
		if (!_currentThread->isInTrivialDownCall)
#endif /* JAVA_SPEC_VERSION >= 21 */
		{
			VM_VMAccess::inlineExitVMToJNI(_currentThread);
		}
```
Can you explain how this benefits JIT?

Reply:
FYI: @tajila

Reply:
For trivial calls there is a performance benefit to be gained by not releasing VMAccess, for the same reason that fastJNI/cINLs are faster than standard JNI. 

Currently we only have and interpreter path, however, in the future the we will have a JIT path as well. For consistency, the interpreter should behave in the same manner as how the JIT will behave in this case. Similar to how, cINLs and fastJNI are consistent with one another in this regard. I believe this is what the comment above is referring to.

perhaps it could just be:
```
/* Only exit to JNI for non-trivial downcalls for better performance. */
```

Reply:
I think Keiths suggestion is equivalent and saves us the `#else`

Reply:
Updated the code and comment as suggested above.